app/src/main/java/ch/epfl/sweng/erpa/services/GameService.java

```diff
@@ -3,15 +3,89 @@
 import com.annimon.stream.Optional;
 
 import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
 
 import ch.epfl.sweng.erpa.model.Game;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
 
 public interface GameService {
     String GAME_UUID_PREFIX = "game|";
     String PROP_INTENT_GAMEUUID = "game uuid";
 
     Optional<Game> getGame(String gid);
+
     Set<Game> getAll();
+
     void saveGame(Game g);
+
     boolean removeGames();
+
+    @NoArgsConstructor
+    @Getter
+    class StreamRefiner {
```

Comment:
Didn't Roos say to avoid default implementations in implementations due to Jacoco?